CPDShell/power.py

```diff
@@ -0,0 +1,39 @@
+from pathlib import Path
+
+from CPDShell.shell import CPContainer
+
+
+class Power:
+    def __init__(self, root_path: Path):
+        self.root_path = root_path
+
+    def __read_all_csv(self):
+        """Iterator for reading all CSV files in the specified directory and its subdirectories."""
+        for csv_file in self.root_path.rglob("*.csv"):
+            data = CPContainer.from_csv(csv_file)
+            yield data
+
+    @staticmethod
+    def __checking_near_point(container: CPContainer):
+        """Checks whether at least one change point is about the expected ones."""
+        margin = 50
+        detected_points = container.result
+        expected_points = container.expected_result
+
+        if not expected_points:
+            return False
+
+        for detected in detected_points:
+            if any(expected - margin <= detected <= expected + margin for expected in expected_points):
```

Comment:
It seems at least this `if` can be turned into boolean `return`. You can expand this further if you want, but that would probably degrade readability already. The main thing is to be careful with empty cases.